src/fields.jl

```diff
@@ -52,3 +60,233 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+function fill_halos(data, grid::Grid1D{Periodic}, LX::Type{Centre})
```

Comment:
What does this function do? The operation of filling up the halos does not require any data input. It just requires the knowledge of the boundary conditions so as to fill up the halos in the right manner.

A docstring would help alleviate any ambiguities perhaps. Or if there are things you feel we disagree in a conceptual level then we can discuss about them either here or over zoom; whatever is easier.

Reply:
I'll add the docstrings. Regarding the functionality of fill_halos, let's discuss it on the above conversation.